tests/Doctrine/Tests/ORM/Functional/Ticket/DDC6303.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Tests\Models\Issue5989\Issue5989Employee;
+use Doctrine\Tests\Models\Issue5989\Issue5989Secretary;
+use Doctrine\Tests\Models\Issue5989\Issue5989Manager;
+use Doctrine\Tests\Models\Issue5989\Issue5989Person;
+
+/**
+ * @group DDC6303
+ */
+class DDC6303Test extends \Doctrine\Tests\OrmFunctionalTestCase
+{
+    public function setUp()
+    {
+        $this->useModelSet('issue5989');
+        parent::setUp();
+    }
+    
+    public function testMixedTypeHydratedCorrectlyInJoinedInheritance()
+    {
+        $manager = new Issue5989Manager();
+
+        $managerTags = ['tag1', 'tag2'];
+        $manager->tags = $managerTags;
+        $this->_em->persist($manager);
+
+        $employee = new Issue5989Employee();
+
+        $employeeTags =['tag2', 'tag3'];
+        $employee->tags = $employeeTags;
+        $this->_em->persist($employee);
+
+        $secretary = new Issue5989Secretary();
+
+        $secretaryTags = 'single_tag';
+        $secretary->tags = $secretaryTags;
```

Comment:
The fact that you use a different type for a property in the inheritance tree is quite confusing: is this intentional?